src/librustc/middle/ty.rs

```diff
@@ -2059,18 +2057,44 @@ pub struct ExistentialBounds<'tcx> {
     pub projection_bounds: Vec<PolyProjectionPredicate<'tcx>>,
 }
 
-#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
-pub struct BuiltinBounds(EnumSet<BuiltinBound>);
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+pub struct BuiltinBounds {
+    bits: u8
+}
 
 impl BuiltinBounds {
-       pub fn empty() -> BuiltinBounds {
-        BuiltinBounds(EnumSet::new())
+    pub fn empty() -> BuiltinBounds {
+        BuiltinBounds { bits: 0 }
+    }
+
+    pub fn insert(&mut self, bound: BuiltinBound) {
+        self.bits = match bound {
```

Comment:
Yeah good point some of these are probably just my functional programmer side coming through ;)